### DISCUSSION TITLE
How to disable right-click menu?

### DISCUSSION THREAD

**Discussion by vitaminace33** [emacs-lsp/lsp-mode#3667] [Q&A] [answered]
Hi there!

Since I have enabled lsp-mode, right-clicking a buffer shows a menu. I would like to revert it to its usual behavior that allows me select and cut. How can I achieve this?

BTW, I am on emacs-v28.1, spacemacs/develop.

Thx in advc.

**Comment by yyoncho** [comment, marked as answer]
```
(with-eval-after-load 'lsp-mode
  (define-key lsp-mode-map (kbd "<mouse-3>") nil))
```

**Reply by vitaminace33** [reply]
Thank you, Ivan. I put it in `dotspacemacs/user-config` and worked flawlessly.

Since we are at it, can this be filtered by major mode? That is, deactivate it in specific modes like (La)TeX while keeping it in C/C++, or maybe the converse, activate it only in specific modes. I don't know what makes more sense.

Anyhow, thx again for the snippet!

**Reply by yyoncho** [reply]
You can bind lsp-mouse-click in `<major>-mode-map` and it will work only for that mode.

**Reply by vitaminace33** [reply]
I see... nice! Thx again.
